### PR TITLE
waypoint support ProxyMetadata

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

### CODE REVIEW

File: manifests/charts/istio-control/istio-discovery/files/waypoint.yaml [hunk 1]
```diff
@@ -126,6 +126,12 @@ spec:
         - name: PROXY_CONFIG
           value: |
                  {{ protoToJSON .ProxyConfig }}
+        {{- if .ProxyConfig.ProxyMetadata }}
```
Comment (hanxiaop) [inline manifests/charts/istio-control/istio-discovery/files/waypoint.yaml:129]: Can you explain why do we need to support this in waypoint?

Comment (zirain) [reply manifests/charts/istio-control/istio-discovery/files/waypoint.yaml:129]: this enable all features guard by env, e.g. PROXY_CONFIG_XDS_AGENT.

**Comment by zirain** [comment]
cc @GregHanson

File: manifests/charts/istio-control/istio-discovery/files/waypoint.yaml (hunk 1 above)
Comment (hzxuzhonghu) [reply manifests/charts/istio-control/istio-discovery/files/waypoint.yaml:129]: was this missed at beginning

Comment (zirain) [reply manifests/charts/istio-control/istio-discovery/files/waypoint.yaml:129]: @howardjohn is this by design?

**Review by howardjohn — APPROVED** [review]

**Comment by GregHanson** [comment]
thanks @zirain !

**Comment by linsun** [comment]
nice work @zirain ! this would allow us to set istio.io/rev on waypoint, yeah?

cc @ilrudie @therealmitchconnors

**Comment by therealmitchconnors** [comment]
> nice work @zirain ! this would allow us to set istio.io/rev on waypoint, yeah?
> 
> cc @ilrudie @therealmitchconnors

No, istio.io/rev on waypoints has been supported since 1.19.  This allows users to write arbitrary env vars to the proxy, usually for alpha feature enablement.

**Comment by linsun** [comment]
@therealmitchconnors thank you for the clarification!

**Pull request merged**